File: Data Types/linked list queue.java
import java.io.*;
import java.util.ArrayList;

class QueueNode{
  int value;
  QueueNode next;
  public QueueNode(int value){
    this.value = value;
    this.next = null;
  }
}
class Queue{
  QueueNode head;
  QueueNode tail;
  public boolean isEmpty(){
    return(head == null);
  }
  
  public int peek(){
    return head.value;
  }
  public void push(int value){
    QueueNode newNode = new QueueNode(value);
    if(isEmpty()){
      head = newNode;
      tail = newNode;
    }else{
      tail.next = newNode;
      tail = newNode;
    }
  }
  public void pop(){
    if(isEmpty()){
      return;
    }
    if(head.next == null){
      head = null;
      tail = null;
      return;
    }else{
      head = head.next;
    }
  }
  public ArrayList<Integer> display(){
    QueueNode current = head;
    ArrayList<Integer> list = new ArrayList<Integer>();
    while(current != null){
      list.add(current.value);
      current = current.next;
    }
    return list;
  }
}

class main{
  public static void main(String[] args){
    Queue l = new Queue();
    l.push(1);
    l.push(2);
    l.push(3);
    l.push(4);

    l.pop();
    
    System.out.println(l.display());
    System.out.println(l.peek());
    
  }
}

File: Data Types/linked list queue.py
class QueueNode(object):
    def __init__ (self, d, n = None):
        self.data = d
        self.next = n
        
    def get_data(self):
        return self.data

    def set_next(self, new_next):
        self.next = new_next
        
class Queue(object):
    def __init__(self, head = None, tail = None):
        self.head = head
        self.tail = tail
    def isEmpty(self):
        return (self.head == None)
    def peek(self):
        return self.head.get_data()
    def push(self, value):
        newNode = QueueNode(value)
        if(self.isEmpty()):
            self.head = newNode
            self.tail = newNode
        else:
            self.tail.set_next(newNode)
            self.tail = newNode
    def pop(self):
        if(self.isEmpty()):
            self.head = None
            self.tail = None
            return
        else:
            self.head = self.head.next
    def display(self):
        arr = []
        current = self.head
        while(current != None):
            arr.append(current.get_data())
            current = current.next
        return arr

def main():
    l = Queue()
    l.push(1)
    l.push(2)
    l.push(3)
    l.push(4)
    
    l.pop()
    l.peek()
    
    arr = l.display()
    
    for j in range(len(arr)):
        print(arr[j])
    
if __name__ == "__main__":
    main()
